Tribler/Core/Modules/gigachannel_manager.py

```diff
@@ -99,12 +106,19 @@ def service_channels(self):
         except Exception:
             self._logger.exception("Error when checking for channel updates")
 
+        if not self.processing:
```

Comment:
Why not put

```python
        try:
            self.check_channels_updates()
        except Exception:
            self._logger.exception("Error when checking for channel updates")
```

in this `if` statement as well?

Reply:
because this is just a normal program flow. If we are processing, we just want to skip re-initializing processing this time.

Reply:
On a second thought, it will really be a little bit safer for keeping the loop alive. I will wrap the call into try/except.

Reply:
But why check for channel updates if we are busy processing anyway? Seems like wasted CPU and I/O.

Reply:
At one point in time, I had this additional check at the beginning of the loop. Then I removed it because the loop runs every 5 seconds, and I wanted the thing to keep the queue full. You're right, I can bring the check back to save some DB i/o.